src/safe_queue.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023 Cisco Systems, Inc. and others.  All rights reserved.
+ */
+#pragma once
+
+#include <mutex>
+#include <optional>
+#include <queue>
+#include <unistd.h>
+
+namespace qtransport {
+
+/**
+ * @brief safeQueue is a thread safe basic queue
+ *
+ * @details This class is a thread safe wrapper for std::queue<T>.
+ * 		Not all operators or methods are implemented.
+ *
+ * @todo Implement any operators or methods needed
+ */
+template <typename T> class safeQueue {
+public:
+  /**
+   * @brief safeQueue constructor
+   *
+   * @param limit     Limit number of messages in queue before push blocks. Zero
+   *                  is unlimited.
+   */
+  safeQueue(uint32_t limit = 0) { this->limit = limit; }
+
```

Comment:
limit = 0 as default is problematic .. may be we should have say video objects for 5 seconds worth of data